### PR TITLE
Adapt documentation of the core installation

### PR DESCRIPTION
**Pull request by bouthilx**
Why:

Based on issue #132 it looks like the title of the section is
misleading. Also the PyPI link was not added now that we have one.

How:

Change title to `installation` and explain how to install the `core`
alone, plus write additional doc on how to install plugins.

### CODE REVIEW

File: docs/src/install/core.rst [hunk 1]
```diff
@@ -21,9 +26,10 @@ User's recommended
```
Comment (dmitriy-serdyuk) [inline docs/src/install/core.rst]: Maybe rename this section to "Recommended to all users" and next one to smth like "Recommended to users who participate in development of Orion"

**Comment by bouthilx** [comment]
Build of docs completed and available at https://orion.readthedocs.io/en/fix-install_core_doc/

File: docs/src/install/core.rst (hunk 1 above)
Comment (dmitriy-serdyuk) [reply docs/src/install/core.rst]: > his way is recommended if you want to work with the bleeding edge version of Oríon.

Also, below it turns out that `master` branch is not bleeding edge? Why is this recommended then?

**Comment by bouthilx** [comment]
@ReyhaneAskari and/or @tsirif please have a look at this PR whenever you can. Compiled doc is available [here](https://orion.readthedocs.io/en/fix-install_core_doc/).

File: docs/src/install/plugins.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Oríon is built to be easily extensible. Algorithms and database backends can be implemented in
+external repositories and installed as plugins. The installation process is very simple, you only
+need to install them as you would for Oríon's core (see :doc:`/install/core`). They will be
+automatically detected and available for Oríon at run-time.
+
+Note that active development is currently focused on the core. Therefore only one algorithm plugin
+is available for now.
+
+For more information about how to develop your own plugins, see section `/plugins/base`.
```
Comment (ReyhaneAskari) [inline docs/src/install/plugins.rst]: The link to `/plugins/base` does not exist.

File: docs/src/install/plugins.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Algorithms
+==========
+
+Skopt algorithms
+----------------
+
+`orion.algo.skopt` provides a wrapper for `Bayesian optimizer`_ using Gaussian process implemented
+in `scikit optimize`_. For more information about algorithms configuration and usage please refer to
+`/user/algorithms`.
```
Comment (ReyhaneAskari) [inline docs/src/install/plugins.rst]: The link to `/user/algorithms` does not exist.

File: docs/src/user/pytorch.rst [hunk 4]
```diff
@@ -5,8 +5,8 @@ Simple example
 Installation and setup
 ======================
 
-First, install Oríon follwing :doc:`/installing` and configure the database (:doc:`/database`).
-Then install `pytorch`, `torchvision` and clone the PyTorch
+First, install Oríon follwing :doc:`/install/core` and configure the database
+(:doc:`/install/database`).  Then install `pytorch`, `torchvision` and clone the PyTorch
 [examples repository](https://github.com/pytorch/examples):
```
Comment (ReyhaneAskari) [inline docs/src/user/pytorch.rst]: The hyper link to "examples repository" is not created correctly.

**Comment by bouthilx** [comment]
@ReyhaneAskari It's fixed. Make sure to refresh all page of the doc you visit otherwise you may get the previous version.

File: docs/src/install/core.rst [hunk 5]
```diff
@@ -1,32 +1,38 @@
-****
-Core
-****
+************************
+Installation of the Core
```
Comment (DatCorno) [inline docs/src/install/core.rst]: I would go with another sentencing, more along the lines of `Installation of Orion's core` or something like that.

File: docs/src/install/core.rst [hunk 6]
```diff
@@ -37,9 +43,8 @@ also recommended, but not necessary. Example:
 .. code-block:: sh
 
    <Create and navigate to the directory where local repo will reside>
-   git clone https://github.com/mila-udem/orion.git
+   git clone https://github.com/mila-udem/orion.git --branch develop
    mkvirtualenv -a $PWD/orion orion
-   workon orion
```
Comment (DatCorno) [inline docs/src/install/core.rst]: Why is the removed?

File: docs/src/install/plugins.rst [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+Installation of plugins
+***********************
+
+Oríon is built to be easily extensible. Algorithms and database backends can be implemented in
+external repositories and installed as plugins. The installation process is very simple, you only
+need to install them as you would for Oríon's core (see :doc:`/install/core`). They will be
+automatically detected and available for Oríon at run-time.
+
+Note that active development is currently focused on the core. Therefore only one algorithm plugin
+is available for now.
```
Comment (DatCorno) [inline docs/src/install/plugins.rst]: We also have gradient descent inside the tests, should we move it to its own repo?

File: docs/src/install/core.rst (hunk 6 above)
Comment (bouthilx) [reply docs/src/install/core.rst]: Because mkvirtualenv automatically sources the virtualenv, you don't need workon right after mkvirtualenv.

**Comment by DatCorno** [comment]
@bouthilx I think we can merge this now

**Comment by tsirif** [comment]
Some titles in tree of contents of some documents are not relevant anymore.
They follow the form "User's Guide 104: Database" in general and they exist in `install/database`, `user/executing`, `user/defining`, `user/algorithms`, `plugins/algorithms`, `developer/testing`, `developer/documenting`, `developer/standards`.

Apart from this small stylistic detail, I am in line with the changes done.

**Comment by bouthilx** [comment]
@tsirif If you agree, I'll leave it like that since @ReyhaneAskari will go over all of this and is likely to modify the fixes I would make anyway.
